lib/carrierwave/base64/adapter.rb

```diff
@@ -34,10 +34,53 @@ def mount_base64_uploader(attribute, uploader_class, options = {})
 
           super Carrierwave::Base64::Base64StringIO.new(data.strip, filename)
         end
-        # rubocop:enable Metrics/MethodLength, Metrics/AbcSize
-        # rubocop:enable Metrics/CyclomaticComplexity
-        # rubocop:enable Metrics/PerceivedComplexity
       end
+
+      def mount_base64_uploaders(attribute, uploader_class, options = {})
+        mount_uploaders attribute, uploader_class, options
+
+        if options[:file_name].is_a?(String)
+          warn(
+            '[Deprecation warning] Setting `file_name` option to a string is '\
+            'deprecated and will be removed in 3.0.0. If you want to keep the '\
+            'existing behaviour, wrap the string in a Proc'
+          )
+        end
+
+        define_method "#{attribute}=" do |data_array|
+          return if data_array == send(attribute).to_s
```

Comment:
The upload will just silently fail, if the user sends a plain string to this setter method. This is not a correct behaviour